### PR TITLE
Feat: enhance the apply-once capability

### LINKED ISSUE #4757: The apply-once strategy has inconsistent behavior in the state maintenance phase and the republishing phase
**Describe the bug**

The apply-once strategy will only take effect during the state maintenance phase

**To Reproduce**
1. I deployed this example first, scaler.replicas is set to  1, and the number of copies of the underlying Deployment is 1.
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: apply-once-app
spec:
  components:
    - name: hello-world
      type: webservice
      properties:
        image: oamdev/hello-world
      traits:
        - type: scaler
          properties:
            replicas: 1
  policies:
    - name: apply-once
      type: apply-once
      properties:
        enable: true
        rules:
        - strategy:
            path:
            - spec.replicas
          selector:
            resourceTypes:
            - Deployment

```

2. Then I manually or through HPA changed the number of copies of the underlying Deployment to 4 --- this 4 is the number of drift copies I want to configure, and I don't want it to be managed by KubevelaController

3. Then I modified this example yaml, scaler.replicas was changed to 3, then I expected the rendered Deployment to be 4, and the result used 3

**Expected behavior**

**Screenshots**

**KubeVela Version**

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by zxbyoyoyo**
Signed-off-by: 朱晓兵 <596908030@qq.com>

### Description of your changes

Fixes #4757 

image

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/resourcekeeper/dispatch.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha1"
+
```
Comment (Somefive) [inline pkg/resourcekeeper/dispatch.go:27]: remove L27

**Review by Somefive — COMMENTED** [review]
Generally LGTM. Variable names need some refactor. /cc @fourierr

**Comment by wonderflow** [comment]
please also add docs and examples for this feature enhancement

**Comment by zxbyoyoyo** [comment]
I think the pr main purpose is to standardize the behavior of the controller, so that apply-once can be consistent in the behavior of application updates and state keeping, but the user should not need to know.

**Review by Somefive — APPROVED** [review]

**Pull request merged**